tests/components/light/test_init.py

```diff
@@ -18,479 +18,479 @@
-            white_value="white_val",
-        )
+    assert len(turn_on_calls) == 1
+    call = turn_on_calls[-1]
+
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TURN_ON
+    assert call.data.get(ATTR_ENTITY_ID) == "entity_id_val"
+    assert call.data.get(light.ATTR_TRANSITION) == "transition_val"
+    assert call.data.get(light.ATTR_BRIGHTNESS) == "brightness_val"
+    assert call.data.get(light.ATTR_RGB_COLOR) == "rgb_color_val"
+    assert call.data.get(light.ATTR_XY_COLOR) == "xy_color_val"
+    assert call.data.get(light.ATTR_PROFILE) == "profile_val"
+    assert call.data.get(light.ATTR_COLOR_NAME) == "color_name_val"
+    assert call.data.get(light.ATTR_WHITE_VALUE) == "white_val"
+
+    # Test turn_off
+    turn_off_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TURN_OFF)
+
+    await common.async_turn_off(
+        hass, entity_id="entity_id_val", transition="transition_val"
+    )
 
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
 
-        assert 1 == len(turn_on_calls)
-        call = turn_on_calls[-1]
+    assert len(turn_off_calls) == 1
+    call = turn_off_calls[-1]
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TURN_ON == call.service
-        assert "entity_id_val" == call.data.get(ATTR_ENTITY_ID)
-        assert "transition_val" == call.data.get(light.ATTR_TRANSITION)
-        assert "brightness_val" == call.data.get(light.ATTR_BRIGHTNESS)
-        assert "rgb_color_val" == call.data.get(light.ATTR_RGB_COLOR)
-        assert "xy_color_val" == call.data.get(light.ATTR_XY_COLOR)
-        assert "profile_val" == call.data.get(light.ATTR_PROFILE)
-        assert "color_name_val" == call.data.get(light.ATTR_COLOR_NAME)
-        assert "white_val" == call.data.get(light.ATTR_WHITE_VALUE)
+    assert light.DOMAIN == call.domain
+    assert SERVICE_TURN_OFF == call.service
+    assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    assert call.data[light.ATTR_TRANSITION] == "transition_val"
 
-        # Test turn_off
-        turn_off_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TURN_OFF)
+    # Test toggle
+    toggle_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TOGGLE)
 
-        common.turn_off(
-            self.hass, entity_id="entity_id_val", transition="transition_val"
-        )
+    await common.async_toggle(
+        hass, entity_id="entity_id_val", transition="transition_val"
+    )
 
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
 
-        assert 1 == len(turn_off_calls)
-        call = turn_off_calls[-1]
+    assert len(toggle_calls) == 1
+    call = toggle_calls[-1]
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TURN_OFF == call.service
-        assert "entity_id_val" == call.data[ATTR_ENTITY_ID]
-        assert "transition_val" == call.data[light.ATTR_TRANSITION]
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TOGGLE
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)